python/ray/tune/experiment/trial.py

```diff
@@ -793,11 +793,11 @@ def get_error(self) -> Optional[TuneError]:
         return None
 
     def _handle_restore_error(self, exc: Exception):
+        # For Restoration errors, we only increment the restore failure count
+        # if the number of failures exceeds the restore retry limit.
         if self.temporary_state.num_restore_failures >= int(
             os.environ.get("TUNE_RESTORE_RETRY_NUM", 0)
         ):
-            # Restore was unsuccessful, try again without checkpoint.
-            self.clear_checkpoint()
             self.run_metadata.num_failures += 1
```

Comment:
what are diff between `self.run_metadata.num_failures` and `self.temporary_state.num_restore_failures`?

Reply:
`num_restore_failures` is the number of failed restoration. 
`num_failures` is the number of failures that caused by user/ application code.

because restoration is not a user defined behavior, but some feature we provided. We don't treat restoration failure same as normal application failure. The behavior is, when the program failed due to application, we increment the `num_failures` and trying to restore the application. If the restoration is successful, the program just goes on. If the restoration fail, we will keep on trying to restore but increments the number of `num_restore_failures` by 1. When the `TUNE_RESTORE_RETRY_NUM` restore reaches, we stop restoration, and increment the `num_failures` by another 1.